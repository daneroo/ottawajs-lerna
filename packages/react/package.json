{
  "name": "@daneroo/ottawajs-lerna-react",
  "version": "1.1.8",
  "description": "A react component which depends on @daneroo/ottawajs-lerna-math",
  "author": "Daniel Lauzon <daniel.lauzon@gmail.com>",
  "homepage": "https://github.com/daneroo/ottawajs-lerna#readme",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "main": "dist/qcic-react.js",
  "umd:main": "dist/qcic-react.umd.js",
  "module": "dist/qcic-react.m.js",
  "source": "src/index.js",
  "directories": {
    "lib": "src",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/daneroo/ottawajs-lerna.git"
  },
  "bugs": {
    "url": "https://github.com/daneroo/ottawajs-lerna/issues"
  },
  "@pika/pack": {
    "pipeline": [
      [
        "@pika/plugin-standard-pkg",
        {
          "exclude": [
            "__tests__/**/*"
          ]
        }
      ],
      [
        "@pika/plugin-build-node"
      ],
      [
        "@pika/plugin-build-web"
      ]
    ]
  },
  "scripts": {
    "prepublishOnly": "npm run build",
    "lint": "echo standard not working...",
    "test": "jest --verbose --coverage --silent",
    "test:watch": "jest --watch --runInBand --silent",
    "prebuild": "npm run lint && npm t && rimraf dist",
    "build": "pack build"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.4.2",
    "@babel/preset-react": "^7.0.0",
    "@pika/pack": "^0.3.6",
    "@pika/plugin-build-node": "^0.3.14",
    "@pika/plugin-build-types": "^0.3.14",
    "@pika/plugin-build-web": "^0.3.14",
    "@pika/plugin-standard-pkg": "^0.3.14",
    "@testing-library/jest-dom": "^4.0.0",
    "@testing-library/react": "^8.0.4",
    "babel-core": "^7.0.0-bridge.0",
    "babel-jest": "^24.8.0",
    "jest": "^24.8.0",
    "react": "^16.8.0",
    "react-dom": "^16.8.0",
    "rimraf": "^2.6.2",
    "standard": "^13.0.0"
  },
  "peerDependencies": {
    "react": "^16.8.0"
  },
  "jest": {
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    }
  },
  "standard": {
    "//parser": "babel-eslint",
    "env": {
      "jest": true
    }
  },
  "dependencies": {
    "@daneroo/ottawajs-lerna-math": "^1.1.8"
  },
  "gitHead": "5222271d8daddc71d6d1b991603a63475e744e90"
}
