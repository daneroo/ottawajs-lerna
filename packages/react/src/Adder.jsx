import React, { useState } from 'react';
import { sum } from '@daneroo/ottawajs-lerna-math';

export default function Adder({numbers}) {
  return (
    <div>
      <p>Adding numbers: {JSON.stringify(numbers)}</p>
      <p>Yields: {sum(...numbers)} </p>
    </div>
  );
}


