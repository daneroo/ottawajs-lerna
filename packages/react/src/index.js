import Adder from './Adder'

export {
  Adder
}

// alternative, more concise syntax for named exports
// export { default as Foo } from './Foo'

// or export default - probably not both...
// export default { Simple, Counter, Fetch, FetchDate, Stringify, Welcome }
