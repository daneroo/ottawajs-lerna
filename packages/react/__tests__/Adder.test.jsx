import React from "react";
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect'

import { Adder } from '../src'

afterEach(cleanup);

describe('Adder', () => {
  test('import Adder from "ottawajs-lerna-react"', () => {
    expect(typeof Adder).toBe('function')
  })
  it('renders the component', () => {
    const { asFragment } = render(<Adder numbers={[1,2,3]} />)
    expect(asFragment()).toMatchSnapshot();
   })
})
