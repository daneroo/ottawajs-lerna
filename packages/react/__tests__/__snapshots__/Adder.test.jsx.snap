// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adder renders the component 1`] = `
<DocumentFragment>
  <div>
    <p>
      Adding numbers: [1,2,3]
    </p>
    <p>
      Yields: 6 
    </p>
  </div>
</DocumentFragment>
`;
